### PR TITLE
Made turf contents listing more performant.

### LINKED ISSUE #24549: TGUI Stats panel Alt Click getting stuck when changing tabs
### Exploit Reports

- [X] I confirm this issue is not an exploit. (Required)

### BYOND Version

515.1633

### Issue Description

When you alt click and change tabs on status panel, it goes back to floor (alt click) tab

### What did you expect to happen?

For it to change tabs

### What happened instead?

It changed tabs and went to previous alt click floor tab

### Why is this bad/What are the consequences?

you need to fiddle with tabs

### Steps to reproduce the issue.

Alt Click floor
Change tabs
Watch it go back to floor tab
Click another tab
And now click your intended tab

### When did the problem start happening?

TGUI stat panel or 1633. (I have not played in a long time)

### Extra information

dreamseeker_knOonAjLCB

### Relevant log output/runtime error

_No response_

### PR DESCRIPTION
**Pull request by FunnyMan3595**
## What Does This PR Do
The turf contents listing (AKA listedturf, client obj view, alt-click view) has been replaced with a high-performance version, easily able to handle tiles with thousands of items on them.
Fixed #24549
Fixes images not loading and causing lag.

## Why It's Good For The Game
Performance and bug fixes are good.

## Images of changes
https://github.com/ParadiseSS13/Paradise/assets/98381/6c4e6b47-8d63-422e-b478-c8cd04fceb26

## Testing
See above.

## Changelog
:cl:
tweak: The alt-click view of tile contents has been rewritten to have better performance, especially with large numbers of items on a tile.
fix: The same view should no longer steal your tab, frequently fail to load images, or cause lag when images fail to load.
/:cl:

### CODE REVIEW

File: code/__HELPERS/iconprocs.dm [hunk 1]
```diff
@@ -447,7 +447,7 @@ GLOBAL_LIST_EMPTY(bicon_cache)
  * * sourceonly - if TRUE, only generate the asset and send back the asset url, instead of tags that display the icon to players
  * * extra_clases - string of extra css classes to use when returning the icon string
```
Comment (Contrabang) [inline code/__HELPERS/iconprocs.dm:448]: while you're here
```suggestion
 * * extra_classes - string of extra css classes to use when returning the icon string
```

File: code/controllers/subsystem/SSstatpanel.dm [hunk 2]
```diff
@@ -137,23 +136,40 @@ SUBSYSTEM_DEF(statpanels)
...
+	var/i = 0
 	for(var/atom/turf_item as anything in obj_window.atoms_to_show)
+		// Limit what we send to the client's rendered section.
+		if(i < min_index)
+			i++
+			continue
+		if(i >= max_index)
+			i++
+			continue
+		i++
```
Comment (Contrabang) [inline code/controllers/subsystem/SSstatpanel.dm:153]: I believe this is logically equivalent, and should give us super worthwhile bytecode improvements
```suggestion
		i++
		if(i <= min_index || i > max_index)
			continue
```

File: code/modules/client/client_procs.dm [hunk 3]
```diff
@@ -1230,6 +1230,16 @@
...
 			stat_tab = payload["tab"]
 			SSstatpanels.immediate_send_stat_data(src)
+		if("Listedturf-Scroll")
+			if(payload["min"] == payload["max"])
+				// Not properly loaded yet, send the default set.
+				SSstatpanels.refresh_client_obj_view(src)
+			else
+				SSstatpanels.refresh_client_obj_view(src, payload["min"], payload["max"])
+		if("Statpanel-Debug")
+			log_debug(payload)
```
Comment (Contrabang) [inline code/modules/client/client_procs.dm:1240]: We probably don't want this on live. An href exploit could potentially use this to write anything to debug log, and therefore write with `rustg_log_write`, which to my knowledge, writes to the disk

**Review by Contrabang — CHANGES_REQUESTED** [review]
byond side review, I'm not confident enough in js to accurately say anything about it but it looks good.

**Review by Contrabang — COMMENTED** [review]
dm side looks good

**Review by DGamerL — APPROVED** [review]
Same with contra. Approving DM side

**Review by Burzah — APPROVED** [review]

**Pull request merged**